### PR TITLE
[#2939] Improve (trino-connector): Trino connector Support retrieve catalogManager object from  `io.trino.server.Server`. 

### PR DESCRIPTION
**Pull request by diqiu50**
### What changes were proposed in this pull request?

Support building a Trino with  patch https://github.com/datastrato/Trino/commit/93b91791f9148c1885c5fac4d59cd5960feec685
Store the CatalogManager object with a member of `io.trino.server.Server`. 

### Why are the changes needed?

Fix: #2939 

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

Manually test

### CODE REVIEW

**Comment by jerryshao** [comment]
@yuqi1129 can you please help to review this?

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogInjector.java [hunk 1]
```diff
@@ -127,18 +127,34 @@ public void init(ConnectorContext context) {
...
-      Object metadataManager = metadata;
-      if (isClassObject(metadata, "TracingMetadata")) {
-        metadataManager = getFiledObject(metadata, "delegate");
+      Object catalogManager = null;
+
+      try {
+        // Try to retrieve catalogManager from static filed of
+        // "io.trino.server.Server.trinoCatalogManager"
+        // If the Trino is build with the patch
+        // https://github.com/datastrato/Trino/commit/93b91791f9148c1885c5fac4d59cd5960feec685
```
Comment (yuqi1129) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogInjector.java:136]: This is not an open- source project, and users can not review the code.

Comment (yuqi1129) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogInjector.java:136]: Furthermore, please provide more details about why we need the changes and the usage cases.

Comment (jerryshao) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogInjector.java:136]: If this is a in house fix, maybe we don't need this PR for Gravitino?

**Comment by yuqi1129** [comment]
@diqiu50 
Please resolve the comments if you have time.

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogInjector.java (hunk 1 above)
Comment (diqiu50) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogInjector.java:136]: This is to facilitate our deployment. If we provide users with the internal version of Trino, they can directly use the released Trino connector without the need to modify and recompile it.

Comment (yuqi1129) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogInjector.java:136]: What's the meaning of `they can directly use the released Trino connector without the need to modify and recompile it`, should users need to modify the Trino connector if we use the community version?

Anyway, I don't think we should expose our internal version of Trino to users for any purpose.

**Pull request closed without merging**